### ISSUE TITLE
Dynamically generated MenuItem's result in invalid manu layout

### ISSUE THREAD

**Issue by wieslawsoltes** [AvaloniaUI/Avalonia#4297]
```
    <Menu x:CompileBindings="False">
        <MenuItem Header="_File">
            <MenuItem Header="_New" Command="{Binding OnNew}" CommandParameter="{Binding}" InputGesture="Ctrl+N"/>
            <Separator/>
            <MenuItem Header="_Open..." Command="{Binding Platform.OnOpen}" InputGesture="Ctrl+O"/>
            <Separator/>
            <MenuItem Header="Execu_te Script..." Command="{Binding Platform.OnExecuteScriptFile}"/>
            <Separator/>
            <MenuItem Header="_Recent" Items="{Binding RecentProjects}">
                <MenuItem.Styles>
                    <Style Selector="MenuItem">
                        <Setter Property="Header" Value="{Binding Name}"/>
                        <Setter Property="Command" Value="{Binding $parent[UserControl].DataContext.Platform.OnOpen}"/>
                        <Setter Property="CommandParameter" Value="{Binding Path}"/>
                    </Style>
                </MenuItem.Styles>
            </MenuItem>
            <Separator/>
            <MenuItem Header="E_xit" Command="{Binding Platform.OnExit}" InputGesture="Alt+F4"/>
        </MenuItem>
        <MenuItem Header="_Help">
            <MenuItem Header="_About Core2D" Command="{Binding Platform.OnAboutDialog}"/>
        </MenuItem>
    </Menu>
```

![2020-07-15_20-05-11](https://user-images.githubusercontent.com/2297442/87579706-a31b5580-c6d6-11ea-9ae1-3214a2488a37.gif)

Version: 0.10.0-preview1

**Comment by maxkatz6** [comment]
I will take a look

**Comment by maxkatz6** [comment]
Can reproduce same with Default theme as well
![image](https://user-images.githubusercontent.com/3163374/87623791-344cf500-c6f4-11ea-9237-56cfcd458292.png)

**Comment by Gillibald** [comment]
Without `SubscribeToItems(_items);` in the ctor https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ItemsControl.cs#L350 isn't called for bound items

**Comment by maxkatz6** [comment]
For some reason I can't reproduce it with OverlayPopups = true, only with Win32 popup implementation. Probably because of different available size for popup.

**Comment by maxkatz6** [comment]
Probably problem is with SharedSizeGroup or related
![1](https://user-images.githubusercontent.com/3163374/88093561-2c5fdb80-cb60-11ea-866f-f5484db34f4d.gif)
![2](https://user-images.githubusercontent.com/3163374/88093563-2cf87200-cb60-11ea-9bd0-059bc7935658.gif)

**Comment by danwalmsley** [comment]
seems to be fixed in 0.10.0-preview2
